File: hw1/insert-sort-func.js
module.exports = function insertSort(arr) {
    if (arr.length < 2) {
        return arr;
    }

    for (let i = 1; i < arr.length; i++) {
        let key = arr[i];
        let j = i - 1;

        for (; j >= 0 && arr[j] > key; j--) {
            arr[j + 1] = arr[j];
        }

        arr[j + 1] = key;
    }

    return arr;
};

File: hw1/python/insert_sort_func.py
def insert_sort(arr):
    """

    :type arr: list
    """
    if len(arr) < 2:
        return arr

    for i in range(len(arr)):
        key = arr[i]
        j = i - 1

        while j >= 0 and arr[j] > key:
            arr[j + 1] = arr[j]
            j -= 1

        arr[j + 1] = key

    return arr
